builtin/providers/aws/resource_aws_inspector_assessment_target.go

```diff
@@ -0,0 +1,117 @@
+package aws
+
+import (
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/inspector"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAWSInspectorAssessmentTarget() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsInspectorAssessmentTargetCreate,
+		Read:   resourceAwsInspectorAssessmentTargetRead,
+		Update: resourceAwsInspectorAssessmentTargetUpdate,
+		Delete: resourceAwsInspectorAssessmentTargetDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"arn": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"resource_group_arn": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsInspectorAssessmentTargetCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).inspectorconn
+
+	targetName := d.Get("name").(string)
+	resourceGroupArn := d.Get("resource_group_arn").(string)
+
+	resp, err := conn.CreateAssessmentTarget(&inspector.CreateAssessmentTargetInput{
+		AssessmentTargetName: aws.String(targetName),
+		ResourceGroupArn:     aws.String(resourceGroupArn),
+	})
+	if err != nil {
+		return err
+	}
+	log.Printf("[DEBUG] Inspector Assessment %s created", *resp.AssessmentTargetArn)
+
+	d.Set("arn", *resp.AssessmentTargetArn)
+	d.SetId(*resp.AssessmentTargetArn)
+
+	return resourceAwsInspectorAssessmentTargetRead(d, meta)
+}
+
+func resourceAwsInspectorAssessmentTargetRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).inspectorconn
+
+	resp, err := conn.DescribeAssessmentTargets(&inspector.DescribeAssessmentTargetsInput{
+		AssessmentTargetArns: []*string{
+			aws.String(d.Get("arn").(string)),
+		},
+	})
+
+	if err != nil {
+		if inspectorerr, ok := err.(awserr.Error); ok && inspectorerr.Code() == "InvalidInputException" {
+			return nil
+		} else {
+			log.Printf("[ERROR] Error finding Inspector Assessment Target: %s", err)
+			return err
+		}
+	}
+
+	d.Set("name", *resp.AssessmentTargets[0].Name)
```

Comment:
If the input ARN isn't valid, the error will be caught, if any other error occurs, that should be caught.  I could set to this to the name input, rather than the name response from the API, which is preferred in this instance?